rocketpy/Motor.py

```diff
@@ -0,0 +1,1937 @@
+                # height = float(desc[2])/1000
+                # mass = float(desc[4])
+                # nozzleRadius = diameter/4
+                # throatRadius = diameter/8
+                # grainNumber = grainnumber
+                # grainVolume = height*np.pi*((diameter/2)**2 -(diameter/4)**2)
+                # grainDensity = mass/grainVolume
+                # grainOuterRadius = diameter/2
+                # grainInitialInnerRadius = diameter/4
+                # grainInitialHeight = height
+                thrustSource = points
+                self.burnOutTime = points[-1][0]
+
+        # Create thrust function
+        self.thrust = Function(
+            thrustSource, "Time (s)", "Thrust (N)", self.interpolate, "zero"
+        )
+        if callable(thrustSource) or isinstance(thrustSource, (int, float)):
+            self.thrust.setDiscrete(0, burnOut, 50, self.interpolate, "zero")
+
+        # Reshape curve and calculate impulse
+        if reshapeThrustCurve:
+            self.reshapeThrustCurve(*reshapeThrustCurve)
+        else:
+            self.evaluateTotalImpulse()
+
+        # Define motor attributes
+        # Grain and nozzle parameters
+        self.nozzleRadius = nozzleRadius
+        self.throatRadius = throatRadius
+        self.grainNumber = grainNumber
+        self.grainSeparation = grainSeparation
+        self.grainDensity = grainDensity
+        self.grainOuterRadius = grainOuterRadius
+        self.grainInitialInnerRadius = grainInitialInnerRadius
+        self.grainInitialHeight = grainInitialHeight
+        # Other quantities that will be computed
+        self.massDot = None
+        self.mass = None
+        self.grainInnerRadius = None
+        self.grainHeight = None
+        self.burnArea = None
+        self.Kn = None
+        self.burnRate = None
+        self.inertiaI = None
+        self.inertiaIDot = None
+        self.inertiaZ = None
+        self.inertiaZDot = None
+        self.maxThrust = None
+        self.maxThrustTime = None
+        self.averageThrust = None
+
+        # Compute uncalculated quantities
+        # Thrust information - maximum and average
+        self.maxThrust = np.amax(self.thrust.source[:, 1])
+        maxThrustIndex = np.argmax(self.thrust.source[:, 1])
+        self.maxThrustTime = self.thrust.source[maxThrustIndex, 0]
+        self.averageThrust = self.totalImpulse / self.burnOutTime
+        # Grains initial geometrical parameters
+        self.grainInitialVolume = (
+            self.grainInitialHeight
+            * np.pi
+            * (self.grainOuterRadius**2 - self.grainInitialInnerRadius**2)
+        )
+        self.grainInitialMass = self.grainDensity * self.grainInitialVolume
+        self.propellantInitialMass = self.grainNumber * self.grainInitialMass
+        # Dynamic quantities
+        self.evaluateMassDot()
+        self.evaluateMass()
+        self.evaluateGeometry()
+        self.evaluateInertia()
+        self.evaluateCenterOfMass()
```

Comment:
This contains a lot of repeated code, which can make maintaining it a hard task (you may fix it in one place, while the other place stays wrong). How much of it could be substituted by calling `super().__init__()`?

Reply:
Oops! I had indeed included a `super().__init__()` to the `SolidMotor` and `HybridMotor` child classes (see  549e6f539ce8b3b93fa6f6beae0b83c0510b261f), but it got lost in the refactoring process where we separated this PR from other changes. I'll be sure to add it back ASAP.